sdk/python/kfp_tekton/compiler/_tekton_handler.py

```diff
@@ -164,17 +164,60 @@ def _handle_tekton_custom_task(custom_task: dict, workflow: dict, recursive_task
     nested_custom_tasks = []
     custom_task_crs_namelist = [custom_task_key for custom_task_key in custom_task.keys()]
     for custom_task_key in custom_task.keys():
         for inner_task_name in custom_task[custom_task_key]['task_list']:
             inner_task_cr_name = '-'.join(group_names[:-1] + [inner_task_name])
             if inner_task_cr_name in custom_task_crs_namelist:
-                nested_custom_tasks.append({"main_ct": custom_task_key, "nested_custom_task": inner_task_cr_name})
+                nested_custom_tasks.append({
+                    "father_ct": custom_task_key,
+                    "nested_custom_task": inner_task_cr_name
+                })
+    # Summary out all of the nested tasks relationships.
     for nested_custom_task in nested_custom_tasks:
+        father_ct_name = nested_custom_task['father_ct']
+        relationships = find_ancestors(nested_custom_tasks, father_ct_name, [], father_ct_name)
+        nested_custom_task['ancestors'] = relationships['ancestors']
+        nested_custom_task['root_ct'] = relationships['root_ct']
+
+    for nested_custom_task in nested_custom_tasks:
+        nested_custom_task_spec = custom_task[nested_custom_task['nested_custom_task']]['spec']
         for custom_task_cr in custom_task_crs:
-            if custom_task_cr['metadata']['name'] == nested_custom_task['main_ct']:
-                custom_task_cr['spec']['pipelineSpec']['tasks'].append(custom_task[nested_custom_task['nested_custom_task']]['spec'])
+            if custom_task_cr['metadata']['name'] == nested_custom_task['father_ct']:
+                # handle parameters of nested custom task
+                params_nested_custom_task = nested_custom_task_spec['params']
+                # nested_custom_task_special_params = the global params that doesn't defined in parent custom task
+                nested_custom_task_special_params = [
+                    param for param in params_nested_custom_task
+                    if '$(params.' in param['value'] and not bool([
+                        True for ct_param in custom_task_cr['spec']['pipelineSpec']['params']
+                        if param['name'] in ct_param['name']
+                    ])
+                ]
+                custom_task_cr['spec']['pipelineSpec']['params'].extend([
+                    {'name': param['name'], 'type': 'string'}for param in nested_custom_task_special_params
+                ])
+                if nested_custom_task['ancestors']:
+                    for custom_task_cr_again in custom_task_crs:
+                        if custom_task_cr_again['metadata']['name'] in nested_custom_task[
+                            'ancestors'] or custom_task_cr_again['metadata']['name'] == nested_custom_task['root_ct']:
+                            custom_task_cr_again['spec']['pipelineSpec']['params'].extend([
+                                {'name': param['name'], 'type': 'string'}for param in nested_custom_task_special_params
```

Comment:
maybe we need to sort the parameters here, in Python 3.6. the parameters generated is out of order which could break our current Travis tests